### DISCUSSION TITLE
[Docs] Talk about faster-cpython in 3.11's What's New?

### DISCUSSION THREAD

**Discussion by Fidget-Spinner** [faster-cpython/ideas#312] [General]
We're two months away from Python 3.11's feature freeze. I was wondering if all the faster-cpython optimizations could be summarised and documented in the 3.11's What's New document as a separate section.?

I volunteer to write the docs if no one else has already (and coordinate efforts if need be). If there's someone else already on this, I hope I can help. I've been[ keeping track of](https://github.com/Fidget-Spinner/cpython_optimization_notes) *most* of the faster-cpython work. Some parts I'm not familiar with and need help for are the ``try-except`` optimization, and deepfreeze/static objects.

The final result will likely be short with little technospeak, but hopefully enough to illustrate exactly *which* parts of their code sped up. I really want to excite users with the new features :).

Subsequent optimizations can be appended to the section as we go along. IMO, it's good to start early to get more reviews. Also, some people (and tech news sites) are already looking towards the What's New in 3.11 document to search for faster-cpython's results. The current document doesn't really help.

The biggest con I can think of is if there's a sudden overhaul of optimizations we need to rewrite. Hence, I'll try to write the docs without focusing too much on the underlying implementation.

WDYT?

**Comment by JelleZijlstra** [comment]
Strongly agree. I think this will also be a great tool to help motivate people to upgrade to 3.11—if I can point to the What's New and say "3.11 will be 20% faster", that will help me (and probably others) get time allocated to do an upgrade.

**Comment by kumaraditya303** [comment]
+1, I volunteer for docs for deepfreeze if @gvanrossum agrees.

**Comment by gvanrossum** [comment]
Sounds like a good idea. I recommend that the three of you get together and draft some text that satisfies your objectives.

**Comment by brandtbucher** [comment]
Sounds like a good plan. Perhaps wait until the feature freeze to actually commit the new docs, though? That way we don't have to remember to keep them in sync over the next few months.

**Comment by iritkatriel** [comment]
Maybe a blog like this one?

https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-6/

**Reply by Fidget-Spinner** [reply]
That's a great blog post. However, I don't think it will fit in the whats new. I hope to keep things under 2 A4 pages (arbitrary, I know :) for the average Python user to appreciate all the work done.

An in-depth technical dive can be a separate document entirely (in someone's personal blog or Github).

**Comment by Fidget-Spinner** [comment]
The work branch is up at https://github.com/Fidget-Spinner/cpython/tree/faster_cpython_whatsnew. I've enabled issues and PRs on that repo to get feedback from Jelle and Kumar Aditya.

**Comment by Fidget-Spinner** [comment]
Done for now in https://github.com/python/cpython/commit/9ffe47df5468a72603f730eae48c2fd4ec615ffa. Seems closing the discussion isn't a thing.
